barretenberg/cpp/src/barretenberg/bb/main.cpp

```diff
@@ -134,7 +137,16 @@ bool verify(const std::string& proof_path, bool recursive, const std::string& vk
     auto acir_composer = new acir_proofs::AcirComposer(MAX_CIRCUIT_SIZE, verbose);
     auto vk_data = from_buffer<plonk::verification_key_data>(read_file(vk_path));
     acir_composer->load_verification_key(barretenberg::srs::get_crs_factory(), std::move(vk_data));
-    auto verified = acir_composer->verify_proof(read_file(proof_path), recursive);
+
+    // If the number of public inputs is 0, then read_file will trigger a failure
+    // because the file will be empty.
+    auto public_inputs_path = proof_path + "-public_inputs";
+    std::vector<uint8_t> pub_inputs_file;
+    if (vk_data.num_public_inputs != 0) {
+        pub_inputs_file = read_file(public_inputs_path);
+    }
+    auto proof_path_file = read_file(proof_path);
+    auto verified = acir_composer->verify_proof_splitted(pub_inputs_file, proof_path_file, recursive);
```

Comment:
This works though note that we are not testing for the case where are public inputs in CI because we only test 1_mul which has no public inputs, the other acir tests will do `proveAndVerify` which does not test the separate workflow of proving _and then_ verifying